eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/LedgerSpec.hs

```diff
@@ -89,3 +106,83 @@ spec = do
+
+  it "Fails predicate when treasury withdrawal has nonexistent return address" $ do
+    modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 2
+    policy <- getGovPolicy
+    protVer <- getProtVer
+    registeredRewardAccount <- registerRewardAccount
+    netId <- use (impGlobalsL . to networkId)
+    unregisteredRewardAccount <- RewardAccount netId . KeyHashObj <$> freshKeyHash
+    deposit <- getsNES $ nesEsL . curPParamsEpochStateL . ppGovActionDepositL
+    anchor <- arbitrary
+    let mkProposal govAction rewardAccount =
+          ProposalProcedure
+            { pProcDeposit = deposit
+            , pProcReturnAddr = rewardAccount
+            , pProcGovAction = govAction
+            , pProcAnchor = anchor
+            }
+        mkTreasuryWithdrawal rewardAccount =
+          TreasuryWithdrawals (Map.singleton rewardAccount $ Coin 1_000) policy
+        validTreasuryWithdrawal =
+          mkProposal (mkTreasuryWithdrawal registeredRewardAccount) registeredRewardAccount
+        invalidTreasuryWithdrawal =
+          mkProposal (mkTreasuryWithdrawal unregisteredRewardAccount) registeredRewardAccount
+
+    if HF.bootstrapPhase protVer
+      then do
+        submitFailingProposal
+          validTreasuryWithdrawal
+          [injectFailure $ DisallowedProposalDuringBootstrap validTreasuryWithdrawal]
+        submitFailingProposal
+          invalidTreasuryWithdrawal
+          [ injectFailure $ DisallowedProposalDuringBootstrap invalidTreasuryWithdrawal
+          , injectFailure $ ConwayReturnAddressDoesNotExist unregisteredRewardAccount
+          ]
+      else do
+        submitProposal_ validTreasuryWithdrawal
+        submitFailingProposal
+          invalidTreasuryWithdrawal
+          [ injectFailure $ ConwayReturnAddressDoesNotExist unregisteredRewardAccount
+          ]
+
+  it "Fails predicate when proposal deposit has nonexistent return address" $ do
+    modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 2
+    policy <- getGovPolicy
+    protVer <- getProtVer
+    registeredRewardAccount <- registerRewardAccount
+    netId <- use (impGlobalsL . to networkId)
+    unregisteredRewardAccount <- RewardAccount netId . KeyHashObj <$> freshKeyHash
+    deposit <- getsNES $ nesEsL . curPParamsEpochStateL . ppGovActionDepositL
+    anchor <- arbitrary
+    let mkProposal govAction rewardAccount =
+          ProposalProcedure
+            { pProcDeposit = deposit
+            , pProcReturnAddr = rewardAccount
+            , pProcGovAction = govAction
+            , pProcAnchor = anchor
+            }
+        mkTreasuryWithdrawal rewardAccount =
+          TreasuryWithdrawals (Map.singleton rewardAccount $ Coin 1_000) policy
+        proposalWithValidReturnAddress =
+          mkProposal (mkTreasuryWithdrawal registeredRewardAccount) registeredRewardAccount
+        proposalWithInvalidReturnAddress =
+          mkProposal (mkTreasuryWithdrawal registeredRewardAccount) unregisteredRewardAccount
```

Comment:
Why are using a governance action that can't be proposed during bootstrap? It would be much easier to use something like InfoAction instead
```suggestion
    let mkProposal govAction rewardAccount =
          ProposalProcedure
            { pProcDeposit = deposit
            , pProcReturnAddr = rewardAccount
            , pProcGovAction = InfoAction
            , pProcAnchor = anchor
            }
        proposalWithValidReturnAddress = mkProposal registeredRewardAccount
        proposalWithInvalidReturnAddress = mkProposal unregisteredRewardAccount
```